src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -1144,5 +1144,16 @@ public override int GetHashCode()
 
             return referenceCulture.CompareInfo.IsPrefix(this, value, ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None);
         }
+
+        [Pure]
+        public Boolean StartsWith(Char value)
```

Comment:
Maybe I should just write
public bool StartsWith(char value) => m_firstChar == value;

Since:

        // For empty strings, this will be '\0' since
        // strings are both null-terminated and length prefixed
        [NonSerialized] private char m_firstChar;